utils/dataset_manifest/core.py

```diff
@@ -182,35 +194,41 @@ def step(self):
     def step(self, value):
         self._step = int(value)
 
-    def __iter__(self):
-        sources = (i for i in self._sources)
-        for idx in range(self._stop):
-            if idx in self.range_:
-                image = next(sources)
-                img = Image.open(image, mode='r')
+    def _get_img_properties(self, image: Union[str, BytesIO]) -> Dict[str, Any]:
+        img = Image.open(image, mode='r')
+        if self._data_dir:
+            img_name = os.path.relpath(image, self._data_dir)
+        else:
+            img_name = os.path.basename(image) if isinstance(image, str) else image.filename
```

Comment:
The `BytesIO` class doesn't have the `filename` member.

Reply:
Fixed types